src/app/ReadHandler.cpp

```diff
@@ -128,6 +128,8 @@ void ReadHandler::ResumeSubscription(CASESessionManager & caseSessionManager,
     // Ask IM engine to start CASE session with subscriber
     ScopedNodeId peerNode = ScopedNodeId(subscriptionInfo.mNodeId, subscriptionInfo.mFabricIndex);
     caseSessionManager.FindOrEstablishSession(peerNode, &mOnConnectedCallback, &mOnConnectionFailureCallback);
+    // Notify the observer that a subscription has been resumed
+    mObserver->OnSubscriptionEstablished(this);
```

Comment:
This does not seeem right.  The order of this during normal subscription setup would be:

1. SetStateFlag(ReadHandlerFlags::ActiveSubscription) (which never seems to happen on the resumption codepath @jtung-apple; should it?).
2. `OnSubscriptionEstablished()` callbacks.
3. `MoveToState(HandlerState::CanStartReporting)`

There is no long time period there where OnSubscriptionEstablished has been called but we are not in CanStartReporting state.

Maybe it's OK to have such a period, but it seems like it would be safer to not have it so the two codepaths match more closely.  So perhaps this call should actually be in `ReadHandler::HandleDeviceConnected`?

Reply:
> SetStateFlag(ReadHandlerFlags::ActiveSubscription) (which never seems to happen on the resumption codepath @jtung-apple; should it?).

Yeah thank you for noticing this, I think `ReadHandlerFlags::ActiveSubscription` should be set. And on top of that I think we should also call OnSubscriptionEstablished on the appCallback. This so that the app knows when subscription resumes/terminates. Currently the app is getting neither.

On this PR, I agree with moving the OnSubscriptionEstablished call for Observer to `HandleDeviceConnected` to mirror the same behavior as incoming subscription.

Reply:
Moved callbacks to HandleDeviceCOnnected and added flag.